src/Forms/DatetimeField.php

```diff
@@ -189,141 +375,167 @@ public function setValue($value, $data = null)
         // Save value
         $this->value = $value;
 
-        // Shift iso date into timezone before assignment to subfields
-        $timezoneFormatter = $this->getTimezoneFormatter();
-        if ($timezoneFormatter) {
-            $value = $timezoneFormatter->format($timestamp);
-        }
-
-        // Set date / time components, which are unaware of their timezone
-        list($date, $time) = explode(' ', $value);
-        $this->dateField->setValue($date, $data);
-        $this->timeField->setValue($time, $data);
         return $this;
     }
 
-    /**
-     * localised time value
-     *
-     * @return string
-     */
     public function Value()
     {
-        $date = $this->dateField->Value();
-        $time = $this->timeField->Value();
-        return $this->joinDateTime($date, $time);
+        return $this->iso8601ToLocalised($this->value);
+    }
+
+    public function dataValue()
+    {
+        return $this->iso8601ToDataValue($this->value);
     }
 
     /**
-     * @param string $date
-     * @param string $time
-     * @return string
+     * Convert an ISO 8601 localised datetime into a saveable representation.
+     *
+     * @param string $datetime
+     * @return string The formatted date and time, or null if not a valid date and time
      */
-    protected function joinDateTime($date, $time)
+    public function iso8601ToDataValue($datetime)
```

Comment:
See comment above; this method shouldn't be necessary, and we should save $value in ready-to-save-to-db format always.